tests/scenarios/v2-addon-peer-deps-test.ts

```diff
@@ -0,0 +1,67 @@
+import { appScenarios, baseV2Addon, baseAddon } from './scenarios';
+import { PreparedApp } from 'scenario-tester';
+import QUnit from 'qunit';
+import merge from 'lodash/merge';
+
+const { module: Qmodule, test } = QUnit;
+
+appScenarios
+  .map('v2-addon-with-optional-peer-dep-on-v1-addon', project => {
+    let v1Addon = baseAddon();
+
+    v1Addon.pkg.name = 'v1-addon';
+    merge(v1Addon.files, {
+      addon: {
+        'index.js': `export class TrackedAsyncData {}`,
+      },
+    });
+
+    let v2Addon = baseV2Addon();
+
+    merge(v2Addon.files, {
+      'index.js': `export { TrackedAsyncData } from 'v1-addon';`,
```

Comment:
FWIW, this doesn't really seem like an optional peer dependency. 

I doubt it matters for the test though, I don't think we have any custom/specific logic around a peer dependency being optional or not (though it's possible that webpack does?)...

Reply:
>  doesn't really seem like an optional peer dependency.

but, if you don't import `index.js`, you don't need `v1-addon`? (or .... how do optional peer deps work?)

Reply:
An example of using an optional peer dep would be

```js

try {
  await import('thing')
} catch (e) {
  // never mind, go on without it
}
```

Or alternatively, using dependencySatisfies from embroider macros to do a similar thing.

Reply:
ohhh! then I think I need to update this test!
(and my ember-resources stuff)